evalml/automl/automl_search.py

```diff
@@ -394,6 +406,7 @@ def __init__(
         max_time=None,
         patience=None,
         tolerance=None,
+        n_splits=3,
```

Comment:
@ParthivNaresh I don't think we need to add this parameter to `AutoMLSearch`. Isn't it implicit in the `data_splitter`?

Reply:
I was thinking about the use case where a user starts with `search` and decides that while they don't want to pass in their own data splitter, they might want to determine the number of splits. It would be passed into the `make_data_splitter` [below](https://github.com/alteryx/evalml/pull/3141/files#diff-8b4df814812e6f5df24c600bc7f429b036e0e2f78613bfbdcdf0ed3ab86bffd6R613)

Reply:
Makes sense! I'm not opposed to that but maybe that should be its own issue. This API change is not related to the scope of the original issue.

Reply:
Gotcha, understandable. I'm fine with making a data split in `search` and `search_iterative` and leaving `n_splits` alone for another issue